build-packages.sh

```diff
@@ -3,159 +3,64 @@
     fi
 
-    lowerI="$(echo $1 | awk '{print tolower($0)}')"
-    case $lowerI in
+    case "$1" in
         -\?|-h|--help)
-            usage
-            exit 1
-            ;;
-
-        x86)
-            __BuildArch=x86
-            ;;
-
-        x64)
-            __BuildArch=x64
-            ;;
-
-        arm)
-            __BuildArch=arm
-            ;;
-
-        arm64)
-            __BuildArch=arm64
-            ;;
-        debug)
-            __BuildType=Debug
-            ;;
-        release)
-            __BuildType=Release
-            ;;
-        checked)
-            __BuildType=Checked
+        usage
+        exit 1
+        ;;
+        -BuildArch=*)
+        unprocessedBuildArgs="$unprocessedBuildArgs $1"
+        __Arch=$(echo $1| cut -d'=' -f 2)
+        ;;
+        *)
+        unprocessedBuildArgs="$unprocessedBuildArgs $1"
     esac
     shift
 done
 
-# Use uname to determine what the OS is.
-OSName=$(uname -s)
-case $OSName in
-    Linux)
-        __BuildOS=Linux
-        ;;
-
-    Darwin)
-        __BuildOS=OSX
-        ;;
-
-    FreeBSD)
-        __BuildOS=FreeBSD
-        ;;
-
-    OpenBSD)
-        __BuildOS=OpenBSD
-        ;;
-
-    NetBSD)
-        __BuildOS=NetBSD
-        ;;
-
-    SunOS)
-        __BuildOS=SunOS
-        ;;
-
-    *)
-        echo "Unsupported OS $OSName detected, configuring as if for Linux"
-        __BuildOS=Linux
-        ;;
-esac
-
-if [ "$__BuildOS" == "Linux" ]; then
-    if [ ! -e /etc/os-release ]; then
-        echo "WARNING: Can not determine runtime id for current distro."
-        export __DistroRid=""
-    else
-        source /etc/os-release
-        export __DistroRid="$ID.$VERSION_ID-$__BuildArch"
-    fi
-fi
-
-__IntermediatesDir="$__ProjectRoot/bin/obj/$__BuildOS.$__BuildArch.$__BuildType"
-
-# Ensure that MSBuild is available
-echo "Running init-tools.sh"
-$__ProjectRoot/init-tools.sh
-
-    echo "Generating nuget packages for "$__BuildOS
-
-    # Invoke MSBuild
-    $__ProjectRoot/Tools/dotnetcli/dotnet "$__MSBuildPath" /nologo "$__ProjectRoot/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.builds" /verbosity:minimal "/fileloggerparameters:Verbosity=normal;LogFile=$binclashlog" /t:Build /p:__BuildOS=$__BuildOS /p:__BuildArch=$__BuildArch /p:__BuildType=$__BuildType /p:__IntermediatesDir=$__IntermediatesDir /p:BuildNugetPackage=false /p:UseSharedCompilation=false
-
-if [ $? -ne 0 ]; then
-    echo -e "\nAn error occurred. Aborting build-packages.sh ." >> $build_packages_log
-    echo "ERROR: An error occurred while building packages, see $build_packages_log for more details."
+$__ProjectRoot/run.sh build-packages -Project=$__ProjectRoot/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.builds -DistroRid=\${OSRid}-$__Arch -UseSharedCompilation=false -BuildNugetPackage=false $unprocessedBuildArgs
+if [ $? -ne 0 ]
+then
+    echo "ERROR: An error occurred while syncing packages; See build-packages.log for more details."
```

Comment:
Should the error message say "while building packages" instead of "syncing"? Same in a few places below.